src/ScriptEditor/ScriptEditor.ts

```diff
@@ -71,6 +71,14 @@ export class ScriptEditor {
       languageDefaults.addExtraLib(reactTypes, "react.d.ts");
       languageDefaults.addExtraLib(reactDomTypes, "react-dom.d.ts");
     }
+    /**
+     * Ignore these errors in the editor:
+     * - Cannot find module ''. Did you mean to set the 'moduleResolution' option to 'node', or to add aliases to the 'paths' option?(2792)
+     * - Cannot use JSX unless the '--jsx' flag is provided.(17004)
+     */
+    monaco.languages.typescript.typescriptDefaults.setDiagnosticsOptions({
+      diagnosticCodesToIgnore: [2792, 17004],
+    });
```

Comment:
This
Adding something like this gets monaco processing JSX correctly, though it does process JSX in `.js` files too. It looks like that is something hard-coded into typescript that can't easily be changed. (You can see this behaviour if you open a random JS file in VS Code and type `<div>`; it will auto-close the tag and show type information as-if it were JSX.)

```suggestion
      languageDefaults.addExtraLib(reactTypes, "react.d.ts");
      languageDefaults.addExtraLib(reactDomTypes, "react-dom.d.ts");
      languageDefaults.setCompilerOptions({
        ...languageDefaults.getCompilerOptions(),
        jsx: monaco.languages.typescript.JsxEmit.React,
        allowUmdGlobalAccess: true,
      });
      /**
       * Ignore these errors in the editor:
       * - 'React' refers to a UMD global, but the current file is a module. Consider adding an import instead.(2686)
       */
      languageDefaults.setDiagnosticsOptions({
        diagnosticCodesToIgnore: [2686],
      });
    }
```

Reply:
Some problems with this approach:
- We still need to disable 2792.
- When we disable 2792, we can use `monaco.languages.typescript.JsxEmit.ReactJSX` instead of `monaco.languages.typescript.JsxEmit.React`. We don't need to disable 2686 in this case.
- We only need to set compiler options for `monaco.languages.typescript.typescriptDefaults`.

The suggestion of `allowUmdGlobalAccess` is good, though. Thanks.

Reply:
I'm not sure why you still need to disable 2792 with the change I suggested here?

Also, I'm not sure why you want to use `JsxEmit.ReactJSX` (which is the source of needing 2792, I think). Looking at the translated source that is actually generated here, it is generating `React.createElement` calls, which matches the behavior of `JsxEmit.React` not `JsxEmit.ReactJSX`.